app/services/user_session.rb

```diff
@@ -0,0 +1,124 @@
+class UserSession
+  attr_reader :session, :token
+
+  TTL = 4 * 60 * 60
+  @@ttl = TTL
+
+  @@namespace = "user_session"
```

Comment:
Hmmm I do not see class vars used to much. Mind making this a class method instead? I understand their use for caching below but might be easier to read if this is just a getter method